shap/maskers/_text.py

```diff
@@ -106,7 +105,9 @@ def __call__(self, mask, s):
             out = self.post_process_sentencepiece_tokenizer_output(out)
         # replace sequence of spaces with a single space and strip beginning and end spaces
         out = re.sub(r"[\s]+"," ",out).strip()
-
+        # delete extra spaces around separator tokens
+        if self.tokenizer.sep_token != None:
+            out = re.sub("[\s]*" + self.tokenizer.sep_token + "[\s]*", self.tokenizer.sep_token, out)
```

Comment:
Yes, this is because we found the tokenizer added extra tokens when encoding then decoding, if the sentence contained separators </s> with extra space around them.